Feature/mallard db5 fixes

sknorr opened the pull request:
@tomschr Can you look at this? I am a bit unsure about the last three commits, ff0e68c, 0998671, bc07983

Commit:
Mallard XSLT: Fix ID references

Commit:
Mallard XSLT: Fix obvious thinko, article => abstract

Commit:
Mallard XSLT: Remove useless/inconsistent xsl:message

Commit:
Mallard XSLT: add a space before book listing

Commit:
Mallard XSLT: Avoid unnecessary commas

Commit:
Mallard XSLT: Remove unused productid parameter

$productid seems to be unused, removal didn't cause any complications.
Not sure if there was future plan for this to be used, so not quite
sure. In any case, couldn't quite follow the description of the
parameter.

Commit:
Mallard XSLT: Add FIXME about hardcoded e-mail address

Not really great to have that in there. Any better way of finding out,
e.g. via info element in the docs we process?

tomschr commented on daps-xslt/mallard/docbook.xsl:
```diff
@@ -175,7 +163,8 @@
   <xsl:template match="book/article | d:book/d:article">
     <xsl:param name="node" select="."/>
-    <link href="help:{$packagename}/{@id}">
+    <xsl:variable name="id" select="($node/@id|$node/@xml:id)[1]"/>
```
There might be no result at all if there is no `id` attribute. Not sure if this is necessary, but you could extend it as follows:

``` xml
<xsl:variable name="id">
   <xsl:choose>
      <xsl:when test="($node/@id|$node/@xml:id)[1]">
         <xsl:value-of select="($node/@id|$node/@xml:id)[1]"/>
      </xsl:when>
      <xsl:otherwise>
         <xsl:value-of select="generate-id($node)"/>
      </xsl:otherwise>
   </xsl:choose>
</xsl:variable>
```

Same applies for other variables with the same structure.

Commit:
Remove the introductory sentence here for consistency with GNOME Docs

The <desc/> of GNOME's topics always only contains the links to topics,
but no real extra text.

Commit:
Mallard XSLT: 'of the following' is shorter, can be used more widely

Commit:
Mallard XML: Remove credit element entirely

These are just overview pages anyway -- there are no meaningful credits
here. And ahrd-coding them is just bad.

Commit:
WIP stuff for THings
